builder/docker/builder.go

```diff
@@ -172,16 +223,50 @@ func runCommandInContainer(client provision.BuilderDockerClient, evt *event.Even
 	}
 	waiter, err := client.AttachToContainerNonBlocking(attachOptions)
 	if err != nil {
-		return err
+		return "", err
```

Comment:
We should return `cont.ID` both here and after a StartContainer error to allow the container to be removed even after a failure. Another possibility is removing the container inside this function if we are to return an error.